### DISCUSSION TITLE
How to get the count of queued jobs

### DISCUSSION THREAD

**Discussion by chrishiste** [dagster-io/dagster#23868] [Q&A] [answered]
### What's the issue or suggestion?

I'm looking for a way to retrieve the count of queued jobs in Dagster. Specifically, I need to monitor the number of jobs waiting to be processed in the queue. Any guidance or examples would be appreciated.

Thanks!

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by garethbrickman** [comment, marked as answer]
You can query the GraphQL API to get all the runs with a `QUEUED` status and use a script to count them: 

```python
import requests

def get_queued_jobs_count():
    url = "http://<your-dagster-instance>/graphql"
    query = """
    {
      pipelineRunsOrError(filter: {statuses: [QUEUED]}) {
        ... on PipelineRuns {
          results {
            runId
            status
          }
        }
      }
    }
    """
    response = requests.post(url, json={'query': query})
    data = response.json()
    queued_jobs = data['data']['pipelineRunsOrError']['results']
    return len(queued_jobs)

queued_jobs_count = get_queued_jobs_count()
print(f"Number of queued jobs: {queued_jobs_count}")
```
